### DISCUSSION TITLE
Make it easier for colleagues to get started with using the Home Office design system prototypes

### DISCUSSION THREAD

**Discussion by tomyems** [UKHomeOffice/design-system#389] [Ideas]
Add help to the get started section to help designers use the React kit and the Home Office-styled GOV.UK prototyping kit.

Examples include:

- showing today's date
- timeout modals

An example might be something along the lines of https://6284d2bf8846d76927f7d70f--hods.netlify.app/ (though this implementation won't work in production as explained in https://github.com/UKHomeOffice/design-system/pull/388)

**Comment by tomyems** [comment]
Before embarking on this specifically, we should first review the structure and content of the get started instructions. This will make it easier to set up a prototype in the first place and get going more easily, solving common problems.

See the code: https://github.com/UKHomeOffice/design-system/pull/397
And the re-structured guidance: https://62b9691eaf135b73397f5255--hods.netlify.app/get-started

**Reply by daniel-ac-martin** [reply]
It looks pretty good, but I wonder if the 'code examples' should just be sub-sections under 'build your prototype'?

**Reply by tomyems** [reply]
Still needs a fair bit of tweaking, hence working in the open. I was wondering this, the question I had was around how best to deal with the two different prototypes. Given that designers are likely to be only using either the React one or the GOV.UK one, then it might make more sense to have the instructions on separate pages, that is list them twice on the left, once under the Home Office kit, and once under the Home Office-styled GOV.UK kit.

The other thought I had was how much authority they should have, putting them in community assets makes them seem less maintained and less reliable, though perhaps easier to update, than in the main sections. (I don't know if this is desired or not.)

**Comment by tomyems** [comment]
User needs:
As a designer I need enough information to decide which prototype I should use, so that I use the one that meets my needs.
As a designer I need to be able to follow the instructions so that I can efficiently use the prototype without needing help from others.

**Comment by tomyems** [comment]
Perhaps we could copy the code styling from [Install the GOV.UK kit](https://govuk-prototype-kit.herokuapp.com/docs/install/install-the-kit.md) as it makes it a bit more obvious what commands go in the terminal.

**Reply by daniel-ac-martin** [reply]
Yes I think so. (Or something similar.)
Been meaning to do this for a while...
Could make for another good ticket for next week.

**Reply by daniel-ac-martin** [reply]
Issue raised on parent repo: https://github.com/daniel-ac-martin/NotGovUK/issues/557

**Comment by tomyems** [comment]
I received a piece of feedback that an interaction designer didn't know we had a figma file for the design system - it's a bit hidden under community resources. To consider making the prototyping journey start with the figma file

**Comment by tomyems** [comment]
Removed code snippets for now to keep the scope down and changes minimal in interations:
https://6343ce420602f64625a36ed9--hods.netlify.app/get-started

**Comment by tomyems** [comment]
Updated: https://6346c7ee7fc6150bbdf17456--hods.netlify.app/get-started

**Comment by daniel-ac-martin** [comment]
FWIW, in terms of prototyping examples, in the past Karwai built this: https://github.com/karypun/mock-wildcat-service

**Comment by tomyems** [comment]
Updated to reflect that the Home Office kit is now a plugin to the GOV.UK prototype kit: https://6441261708511d1ab7f264b1--hods.netlify.app/get-started

**Comment by tomyems** [comment]
Clarified some of the guidance: https://64415958ef92535902cb28e9--hods.netlify.app/get-started

**Reply by tomyems** [reply]
Comments by Wednesday 3 May, to then be published

**Comment by tomyems** [comment]
Removed experimental from React prototype: https://645100d487de47178922e21a--hods.netlify.app/get-started

**Comment by tomyems** [comment]
https://645368a7ea262d6054bec09b--hods.netlify.app/get-started

Moved styling

**Comment by roblangdongov** [comment]
Hi @tomyems @daniel-ac-martin - first off, you've done a great job with the HO prototype flavour, thank you.

I've just gone through the **Add Home Office plugin to GOV.UK prototype** process, documented here: [https://design.homeoffice.gov.uk/get-started/home-office-plugin](https://design.homeoffice.gov.uk/get-started/home-office-plugin)

A couple of things seem to have been updated in the GOV prototype - which you probably know about, but thought I'd highlight in case you hadn't...

Under **Header, footer and page background** 
The gov URL has changed from {% extends "govuk-prototype-kit/layouts/main.html" %} to
{% extends "govuk-prototype-kit/layouts/govuk-branded.njk" %} (I installed v13.13.4 of the kit)

It still seems to work fine!

The other thing I noticed was a couple of plugin install failure messages AFTER installing the HO kit, those were:
- Task list
- Step by step
- HMRC Front End

Having said that, they seem to have installed fine, so it's just a heads up, I guess.

Hope that helps, as I say, really easy to use, thanks for putting all this together, it's an excellent tool and resource, cheers!
Rob

**Reply by tomyems** [reply]
Ah amazing thanks - okiedokie will look to update the instructions :) (and see what's going wrong with the install error messages)
